src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -27,21 +27,21 @@ public TorpedoStore(int numberOfTorpedos){
       }
     }
   }
-
+  Random generator = new Random();
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
```

Comment:
The - sign is at the wrong side of the = sign.  The compiler will understand it as an assignment of the negative of numberOfTorpedos